hpx/parallel/algorithms/remote/dispatch.hpp

```diff
@@ -175,10 +121,8 @@ namespace hpx { namespace parallel { namespace util { namespace remote
 
 HPX_REGISTER_PLAIN_ACTION_TEMPLATE(
     (template <typename Algo, typename ExPolicy, typename IsSeq, typename R,
-        BOOST_PP_ENUM_PARAMS(N, typename Arg)>),
-    (hpx::parallel::util::remote::BOOST_PP_CAT(algorithm_invoker_action, N)<
-        Algo, ExPolicy, IsSeq, R, BOOST_PP_ENUM_PARAMS(N, Arg)>))
-
-#undef N
+        typename... Args>),
+    (hpx::parallel::util::remote::algorithm_invoker_action<
+        Algo, ExPolicy, IsSeq, R(Args const&...)>))
```

Comment:
Should we include macros to support registration of dispatched functions? This won't work in truly heterogeneous environments otherwise.